Src/FluentAssertions/Common/TypeExtensions.cs

```diff
@@ -178,16 +178,19 @@ public static bool OverridesEquals(this Type type)
     /// <returns>
     /// Returns <see langword="null"/> if no such property exists.
     /// </returns>
-    public static PropertyInfo FindProperty(this Type type, string propertyName, Type preferredType)
+    public static PropertyInfo FindProperty(this Type type, string propertyName)
     {
-        List<PropertyInfo> properties =
-            type.GetProperties(AllInstanceMembersFlag)
-                .Where(pi => pi.Name == propertyName)
-                .ToList();
+        while (type != typeof(object))
+        {
+            if (type.GetProperty(propertyName, AllInstanceMembersFlag | BindingFlags.DeclaredOnly) is { } property)
+            {
+                return property;
+            }
+
+            type = type.BaseType;
+        }
 
-        return properties.Count > 1
-            ? properties.SingleOrDefault(p => p.PropertyType == preferredType)
-            : properties.SingleOrDefault();
+        return null;
     }
```

Comment:
Nice! 🚀 

I was fiddling with this method semi-recently as I didn't like how we called `ToList()` only the select a single member from it.

Reply:
I had completely forgot that, way to many beers ago.
But that explains why I bonded so much with it 😂